src/test/java/frc/robot/subsystems/example/HoodTest.java

```diff
@@ -0,0 +1,36 @@
+package frc.robot.subsystems.example;
+
+import edu.wpi.first.units.Units;
+import frc.robot.subsystems.hood.Hood;
+import frc.robot.subsystems.hood.HoodIO;
+import frc.robot.subsystems.hood.HoodIOSim;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class HoodTest implements AutoCloseable {
+    private final Hood hood;
+    private final double Delta = 0.05;
+
+    public HoodTest() {
+        HoodIO hoodIO = new HoodIOSim();
+        Hood.initialize(hoodIO);
+        hood = Hood.getInstance();
+    }
+
+    @Test
+    public void testSetAngle() throws InterruptedException {
+        hood.setDefaultCommand(hood.setAngle(() -> Units.Radians.of(0.5).mutableCopy()));
+        Thread.sleep(3000);
```

Comment:
Same as the shooter, you need to call the `CommandScheduler.getInstance.run()` before sleeping.

Reply:
I just realized that the update of the simulated motor needs to be called.  I think this means that the test has to run on a different thread.

Reply:
there is an example of how to run the test it in a different thread in what katzuv sent i think
https://github.com/wpilibsuite/allwpilib/blob/main/wpilibjExamples/src/test/java/edu/wpi/first/wpilibj/examples/elevatorsimulation/ElevatorSimulationTest.java